File: app/src/main/java/pro/anton/averin/networking/testrest/navigation/Navigator.java
package pro.anton.averin.networking.testrest.navigation;

import javax.inject.Inject;
import javax.inject.Singleton;

import pro.anton.averin.networking.testrest.rx.RxBus;
import pro.anton.averin.networking.testrest.rx.events.NaviManagerScreenEvent;
import pro.anton.averin.networking.testrest.rx.events.NaviResponseScreenEvent;

@Singleton
public class Navigator {

    @Inject
    RxBus rxBus;

    @Inject
    public Navigator() {

    }

    public void navigateToResponseScreen() {
        rxBus.send(new NaviResponseScreenEvent());
    }

    public void navigateToManagerScreenForSave() {
        rxBus.send(NaviManagerScreenEvent.withSave());
    }

    public void navigateToManagerScreen() {
        rxBus.send(NaviManagerScreenEvent.withoutSave());
    }
}

File: app/src/main/java/pro/anton/averin/networking/testrest/navigation/Navigator.kt
package pro.anton.averin.networking.testrest.navigation

import javax.inject.Inject

import pro.anton.averin.networking.testrest.di.ActivityScope
import pro.anton.averin.networking.testrest.rx.RxBus
import pro.anton.averin.networking.testrest.rx.events.NaviResponseScreenEvent
import pro.anton.averin.networking.testrest.utils.IntentBuilder
import pro.anton.averin.networking.testrest.views.base.BaseActivity
import javax.inject.Named

@ActivityScope
class Navigator
@Inject
constructor() {

    @Inject
    lateinit var baseActivity: BaseActivity
    @Inject
    lateinit var rxBus: RxBus
    @Inject
    lateinit var intentBuilder: IntentBuilder

    fun navigateToResponseScreen() {
        rxBus.post(NaviResponseScreenEvent())
    }

    fun navigateToManagerScreenForSave() {
        baseActivity.startActivity(intentBuilder.intentToManager_withSave())
    }

    fun navigateToManagerScreen() {
        baseActivity.startActivity(intentBuilder.intentToManager_withoutSave())
    }
}
